src/p5.dance.js

```diff
@@ -1179,6 +1189,12 @@ module.exports = class DanceParty {
     }
   }
 
+  setFuncContext(type, key) {
+    this.world.userEventTriggeredType = type;
```

Comment:
I see this in the validation code doc, but unclear to me what it represents? Can you elaborate?

Reply:
Also wondering if this validation could fail sooner if you wanted it to (maybe we don't from a curriculum perspective)?

```
if (nativeAPI.getTime("measures") > 4) {
  // If the Down key is pressed, but the AI block was not called by the key event, then fail immediately.
  if (World.aiBlockContextUserEventKey !== 'down') {
    console.log('failed bc you did not include the AI block in the when Down key pressed block');
     console.log('userEventTriggeredType', World.userEventTriggeredType);
    nativeAPI.fail();
  }
}
```

Reply:
> I see this in the validation code doc, but unclear to me what it represents? Can you elaborate?

Just removed from the validation code - was just a print statement for debugging purposes. And I forgot to remove it from `p5.dance.js` so just deleted. I was planning to use to keep track of the `key` pressed, but then realized we already had a set defined for this purpose! `this.world.keysPressed`. Thanks for the catch of the extraneous code!

Reply:
> Also wondering if this validation could fail sooner if you wanted it to (maybe we don't from a curriculum perspective)?
> 
> ```
> if (nativeAPI.getTime("measures") > 4) {
>   // If the Down key is pressed, but the AI block was not called by the key event, then fail immediately.
>   if (World.aiBlockContextUserEventKey !== 'down') {
>     console.log('failed bc you did not include the AI block in the when Down key pressed block');
>      console.log('userEventTriggeredType', World.userEventTriggeredType);
>     nativeAPI.fail();
>   }
> }
> ```

I believe curriculum would want to give the user some time to press the 'Down' key while the program is running.

Reply:
Got it (I think), just finally wrapping my head around that you're not actually inspecting the blocks themselves now, so an immediate check isn't possible (right?)

Reply:
Yes - that's right!